File: test/language/templates/python/StructFullNameTemplateArgumentTest.py
import zserio

import Templates

class StructFullNameTemplateArgumentTest(Templates.TestCase):
    def testReadWrite(self):
        structFullNameTemplateArgument = self.api.StructFullNameTemplateArgument(
            self.api.TemplatedStruct_Storage_C76E422F(self.api.import_storage.Storage(42)),
            self.api.TemplatedStruct_Storage_A3A4B101(self.api.Storage("string"))
        )

        writer = zserio.BitStreamWriter()
        structFullNameTemplateArgument.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readStructFullNameTemplateArgument = self.api.StructFullNameTemplateArgument()
        readStructFullNameTemplateArgument.read(reader)
        self.assertEqual(structFullNameTemplateArgument, readStructFullNameTemplateArgument)

File: test/language/templates/cpp/StructFullNameTemplateArgumentTest.cpp
#include "gtest/gtest.h"

#include "templates/struct_full_name_template_argument/StructFullNameTemplateArgument.h"

namespace templates
{
namespace struct_full_name_template_argument
{

TEST(StructFullNameTemplateArgumentTest, readWrite)
{
    StructFullNameTemplateArgument structFullNameTemplateArgument;
    structFullNameTemplateArgument.setStructExternal(
            TemplatedStruct_Storage_08C0ED6D{storage::Storage{42}});
    structFullNameTemplateArgument.setStructInternal(
            TemplatedStruct_Storage_A3A4B101{Storage{std::string{"string"}}});

    zserio::BitStreamWriter writer;
    structFullNameTemplateArgument.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    StructFullNameTemplateArgument readStructFullNameTemplateArgument(reader);

    ASSERT_TRUE(structFullNameTemplateArgument == readStructFullNameTemplateArgument);
}

} // namespace struct_full_name_template_argument
} // namespace templates
